store/postgres/src/subgraph_store.rs

```diff
@@ -424,6 +424,21 @@ impl SubgraphStoreInner {
         store.find_layout(site)
     }
 
+    pub(crate) fn deployment_exists(&self, id: &DeploymentHash) -> Result<bool, StoreError> {
+        let (store, site) = match self.store(id) {
+            Ok(pair) => pair,
+            Err(err) => {
+                if let StoreError::DeploymentNotFound(_) = err {
+                    return Ok(false);
+                } else {
+                    return Err(err);
+                }
+            }
```

Comment:
I think we can use a shorter form here:

```suggestion
            Err(StoreError::DeploymentNotFound(_)) => return Ok(false)
            Err(err) => return Err(err)
```